lib/internal/benchmark/histogram.js

```diff
@@ -0,0 +1,206 @@
+'use strict';
+
+const { validateNumber } = require('internal/validators');
+const { codes: { ERR_OUT_OF_RANGE } } = require('internal/errors');
+const {
+  MathMin,
+  MathMax,
+  MathCeil,
+  MathSqrt,
+  MathPow,
+  MathFloor,
+  NumberMAX_SAFE_INTEGER,
+  ArrayPrototypeSort,
+  ArrayPrototypePush,
+  ArrayPrototypeFilter,
+  ArrayPrototypeReduce,
+  ArrayPrototypeSlice,
+  NumberIsNaN,
+  Symbol,
+} = primordials;
+
+const kStatisticalHistogramRecord = Symbol('kStatisticalHistogramRecord');
+const kStatisticalHistogramFinish = Symbol('kStatisticalHistogramFinish');
+
+class StatisticalHistogram {
+  /**
+   * @type {number[]}
+   * @default []
+   */
+  #all = [];
+  /**
+   * @type {number}
+   */
+  #min;
+  /**
+   * @type {number}
+   */
+  #max;
+  /**
+   * @type {number}
+   */
+  #mean;
+  /**
+   * @type {number}
+   */
+  #cv;
+
+  /**
+   * @returns {number[]}
+   */
+  get samples() {
+    return ArrayPrototypeSlice(this.#all);
+  }
+
+  /**
+   * @returns {number}
+   */
+  get min() {
+    return this.#min;
+  }
+
+  /**
+   * @returns {number}
+   */
+  get max() {
+    return this.#max;
+  }
+
+  /**
+   * @returns {number}
+   */
+  get mean() {
+    return this.#mean;
+  }
+
+  /**
+   * @returns {number}
+   */
+  get cv() {
+    return this.#cv;
+  }
+
+  /**
+   * @param {number} percentile
+   * @returns {number}
+   */
+  percentile(percentile) {
```

Comment:
We already have the built-in histogram API... why create another vs. modifying / building off the existing?

Reply:
I tried using it initially, but the current Histogram API didn't support float numbers, so we lost some precision over the results.

Also, we wanna remove the outliers, so we thought was better to create a new Histogram class.

Reply:
I'd prefer modifying the current in a way that adds the needed functionality.

Reply:
The current one is implemented using C++, and we didn't have access to the underlying samples/values https://nodejs.org/api/perf_hooks.html#class-histogram

Maybe can we do a follow-up exposing more methods on the Histogram class, but for now, I think it will not be worth it unless we choose to not have `removeOutliers` and `cv`, then using Histogram class makes total sense.